diff --git a/README.md b/README.md
index 86ad3a3..ba0dea7 100644
--- a/README.md
+++ b/README.md
@@ -93,6 +93,10 @@ return [
 ];
 ```

+### Specifying the path to your language folder
+
+Out of the box, this package points to `/app/lang` in Laravel 4 or the `resources/lang` in Laravel >= 5 but you can specify where yours exists by editing the `lang_path` key in the config file.
+
 ### Using [gulp](http://gulpjs.com/) (optional)

 Install [`gulp-shell`](https://github.com/sun-zheng-an/gulp-shell) and then run it directly in your `gulpfile.js`:
diff --git a/src/Mariuzzo/LaravelJsLocalization/LaravelJsLocalizationServiceProvider.php b/src/Mariuzzo/LaravelJsLocalization/LaravelJsLocalizationServiceProvider.php
index eab4548..01b7b48 100644
--- a/src/Mariuzzo/LaravelJsLocalization/LaravelJsLocalizationServiceProvider.php
+++ b/src/Mariuzzo/LaravelJsLocalization/LaravelJsLocalizationServiceProvider.php
@@ -70,11 +70,16 @@ public function register()

             $files = $app['files'];

-            if ($laravelMajorVersion === 4) {
-                $langs = $app['path.base'].'/app/lang';
-            } elseif ($laravelMajorVersion >= 5) {
-                $langs = $app['path.base'].'/resources/lang';
+            $languagePath = $this->app['config']->get('localization-js')['lang_path'];
+            if (empty($languagePath)) {
+                if ($laravelMajorVersion === 4) {
+                    $languagePath = '/app/lang';
+                } elseif ($laravelMajorVersion >= 5) {
+                    $languagePath = '/resources/lang';
+                }
             }
+            $langs = $app['path.base'].$languagePath;
+
             $messages = $app['config']->get('localization-js.messages');
             $generator = new Generators\LangJsGenerator($files, $langs, $messages);

