--- a/src/Client/Action/ActionApi.php
+++ b/src/Client/Action/ActionApi.php
@@ -69,7 +69,7 @@
 		return $clientResponse;
 	}
 
-	private function getClient(): ClientInterface {
+	public function getClient(): ClientInterface {
 		if ( !$this->client instanceof ClientInterface ) {
 			$clientFactory = new ClientFactory();
 			$clientFactory->setLogger( $this->logger );
--- a/src/Client/Rest/RestApi.php
+++ b/src/Client/Rest/RestApi.php
@@ -68,7 +68,7 @@
 		return $clientResponse;
 	}
 
-	private function getClient(): ClientInterface {
+	public function getClient(): ClientInterface {
 		if ( !$this->client instanceof ClientInterface ) {
 			$clientFactory = new ClientFactory();
 			$clientFactory->setLogger( $this->logger );